app/models/user.rb

```diff
@@ -351,7 +351,10 @@ def can_change_admin_flag?
   def editing_self?(options = {})
     options[:controller].to_s == 'users' &&
       options[:action] =~ /edit|update/ &&
-      options[:id].to_i == self.id
+      options[:id].to_i == self.id ||
+    options[:controller].to_s =~ /api\/v\d+\/users/ &&
```

Comment:
I would rather check for whole match `/\A/api\/v\d+\/users/\Z/`, just to prevent a case, where we would have `users.+` controller in the future - not expecting but still, it's better safe than sorry.